src/ui.navigation-progress/composables/attrs.composable.js

```diff
@@ -5,72 +5,49 @@ import defaultConfig from "../configs/default.config";
 import { computed } from "vue";
 
 export function useAttrs(props) {
-  const { config, getAttrs, getColor, setColor } = useUI(defaultConfig, () => props.config);
-  const { progress, indicator, step } = config.value;
+  const { config, getAttrs, getColor, setColor, isSystemKey } = useUI(
+    defaultConfig,
+    () => props.config,
+  );
+  const attrs = {};
 
-  const cvaProgress = cva({
-    base: progress.base,
-    variants: progress.variants,
-    compoundVariants: progress.compoundVariants,
-  });
+  for (const key in defaultConfig) {
+    if (isSystemKey(key)) continue;
 
-  const cvaIndicator = cva({
-    base: indicator.base,
-    variants: indicator.variants,
-    compoundVariants: indicator.compoundVariants,
-  });
+    let classes = "";
+    let value = config.value[key];
 
-  const cvaStep = cva({
-    base: step.base,
-    variants: step.variants,
-    compoundVariants: step.compoundVariants,
-  });
+    const isVariants = value.variants || value.compoundVariants;
+    const excludeKeys = ["progress", "indicator", "step"];
```

Comment:
why this need?

Reply:
The idea was to separate keys which needed color props and apply setColor function only to them in favor of better optimization. Taking in attention your [next comment](https://github.com/vuelessjs/vueless/pull/64#discussion_r1670622468)  we don't actually need it.